README
NAME
    Handel - A cart/order/checkout framework with AxKit/TT/Catalyst support

SYNOPSIS
        use Handel;
        Handel->config_class('My::ConfigReader');
    
        my $config = Handel->config;
        # $config->isa('My::ConfigReader')

DESCRIPTION
    This is a generic class containing the default configuration used by
    other Handel classes.

    To learn more about what Handel is and how it works, take a look at the
    manual.

METHODS
  config_class
    Arguments: $config_class

    Gets/sets the name of the configuration class to use. The default
    configuration class is Handel::ConfigReader.

    A Handel::Exception exception will be thrown if the specified class can
    not be loaded.

  config
    Returns an instance of the specified configuration class.

SEE ALSO
    Handel::Cart, Handel::Order, Handel::Checkout

AUTHOR
        Christopher H. Laco
        CPAN ID: CLACO
        claco@chrislaco.com
        http://today.icantfocus.com/blog/


Changes
Revision history for Perl module Handel

1.00014 Fri Dec 30 09:59:24 2011
  - Update to latest Class::Accessor::Grouped to fix warnings
  - Fixed various warnings and test failures on recent DBIx::Class

1.00013 Sun Aug 8 17:07:36 2010
	- Fixed live currency conversion compat tests to be author only
	- Fixed decimal comparison under -Duselongdouble compiled Perls
	- Fixed issue with Storage::Result::can aborting when UNIVERSAL::isa and
		Class::XSAccessors are both loaded.

1.00012 Sun Aug 1 13:40:04 2010
    - Upgraded to Test::File::Contents 0.05 (LESTERM)
    - Fixed: timzone in extra => {} is deprecated, its now directly on the
	    columns definition (LESTERM)
	- Added CONTRIBUTORS file
	- Tweaked manifest.t for .git
	- Tweaked spelling tests
	- Updated PREREQ
	- Tested on 5.8.9, 5.10.1, 5.12.1


1.00011 Wed Aug 20 19:32:44 2008
    - Order->create($cart) uses storage->has_column instead of can
    - Removed Build.PL to make Module::Install happy
    - Removed $self->stash->clear call in Checkout->process
    - Checkout->new phases now eval strings in arrayref correctly
    - Deprecating phase names as constants. Just use strings and be happy.
    - Added tests for plugins add_handler using strings
    - Fixed compat.t tests under 5.10
    - Fixed storage_dbic_clone.t failure under 5.10

1.00010 Sat Mar 1 13:59:23 2008
    - Fixed some perl critic failures
    - Added Session plugins to Catalyst support prereqs
    - Added reminder to add Session plugins to MyApp in starter

1.00009 Sat Jan 12 22:14:22 2008
    - Cart->restore does update instead of quantity in case autoupdates
        are disabled
    - Cart->restore(REPLACE) now checks to make sure name/description exists

1.00008 Sun Jan 6 17:36:23 2008
    - Storage::DBIC search/search_items now assume that any filter
        column is me. if there is no . in the name. This is a fix
        for Mango, which joins pricing for wishlists and a search on
        id yields ambiguous column name errors.

1.00007 Tue Jan 1 12:16:23 2008
    - Fixed instance of using blessed without loading Scalar::Util
    - Use DBIx::Class 0.08008 and set default DATETIME timezones to UTC

1.00006 Sun Oct 21 13:10:23 2007
    - Updated Clone prereq version to 0.28 (Collin Condray)
    - Fixed Quicktart to use schema --install (Charlie Garrison)
    - Fixed various spelling errors and critic checks
    - Added test skips for DBD::SQLite 1.14

1.00005 Wed Aug 22 21:42:33 2007
    - Catalyst Helper code uses as_string instead of format
    - Updated profiles regexps for newer YAML

1.00004 Tue Aug 21 21:01:10 2007
    - Removed RDBO M::I feature in Makefile.PL (RT#28461,ANDK)
    - Fixed exception tests message checks under l10n (RT#28927,RDROUSIES)
    - Added Bulgarian L10N - Kliment A. Ognianov

1.00003 Tue July 24 20:55:32 2007
    - Changed tests to use TEST_AUTHOR instead
    - Converted to Module::Install

1.00002 Sat June 30 17:15:12 2007
    - Catalyst Model tests now skip when Catalyst not installed
    - Fixed Catalyst tests after fresh perl install explosion
    - Fixed pod coverage

1.00001 Mon June 25 22:05:26 2007
    - Fixed basic.t use_ok failures due to dev install
    - Fixed various test skip counts
    - Fixed Component::Validation to deal with new
        DBIC::Validation/throw_exception changes

1.00000 Sun June 24 08:57:22 2007
    - constraint_price now uses $currency->value if value isa
        Data::Currency object

0.99_19 Thur June 21 21:02:36 2007
    - Storage->setup now puts non-accessor keys into %$self
    - Removed incomplete L10N Lexicons
    - Cleanup coverage before 1.0 release
    - DBIx::Class 0.08 Released & Required
    - Added AutoCommit to Handel::Test
    - Tweaked failing checkout process tests

0.99_18 Wed Apr 18 19:55:09 2007
    - Added --version to handel.pl/handel.bat
    - Fixed Handel::Compat::Currency pod
    - Updated prereq to use DBIx::Class 0.07999_02
    - Added skips for DBD::SQLite 1.13 (teh devil) causing failing tests
    - Updated Perl::Critic tests

0.99_17 Wed Mar 21 20:11:24 2007
    - Renamed ConfigReaders %Defaults to %DEFAULTS
    - Removed t/style_critic_tests. ETOOMUCHWORKFORLITTLEBENEFIT
    - Updated Test::MockObject in tests to 1.07
    - Fixed destroy in Order
    - Fixed has_column check when setting item currency columns

    BREAKING API CHANGE:
    - Handel::Currency isa Data::Currency
    - H::C::format moved to as_string; now just returns the format

0.99_16 Thur Dec 21 16:21:15 2006
    - Added \%options/order_by to search/items in Cart/Order
    - Added has_column to DBIC Storage to use ResultSource->has_column
    - Added has_column to Result/DBIC Result
    - Fixed TT demo code. Maybe that will die some day.
        Seriously. Two words: Catalyst Helpers. :-)
    - Fixed issue where Compat Currency convert failed with converter returns 0
    - Changed quickstart setup script to *_handel.pl
    - Added 0700 perms to handel.pl app script
    - Storage::process_error now only dies Handel::Exception subclasses instead
        of all blessed objects
    - Renamed DBIC schema classes into Schema::DBIC space, like their RDBO
        cousins

0.99_15 Sun Nov 12 17:50:52 2006
    - L10N files now use constant-like keys rather than english errors
    - L10N::es_es.pm reworked by Diego Kuperman
    - L10N::fr reworked by Pierrick DINTRAT
    - Added l10n_lexicon_synced.t to ensure language lexicons are synced with
        each other via key checks
    - Removed constraint_uuid check from Checkout order/cart since it's not
        subclass friendly (RT#19813,TKP)
    - Checkout cart now throws an exception of now order can be found matching
        the search criteria
    - Converted tests to use Handel::Test
    - Cleanup tests against Perl::Critic
    - Fixed Checkout prototype mismatch warning under perl > 5.8.8
    - Added more config tests for MP1/MP2
    - Added more Schema connect tests
    - Added more Base tests
    - Added Iterator tests

Makefile.PL
# $Id$
use strict;
use warnings FATAL => 'all';
use inc::Module::Install 0.65;

if ($ENV{TEST_HTTP}) {
    eval 'use Apache::Test 1.27';
    if ($@) {
        print "Apache-Test >= 1.27 is required!\n";
        tests 't/*.t';
    } else {
        tests join(' ', glob('t/*.t'));

        require Apache::TestMM;
        Apache::TestMM->import(qw(test clean));

        my @defines;
        eval 'use Apache::Template';
        if (!$@) {
            push @defines, 'TT2';
        };

        ## for whatever reason, use AxKit causes issues
        ## use-ing A::A::Exception doesn't
        eval 'use Apache::AxKit::Exception';
        if (!$@) {
            push @defines, 'AXKIT';
        };

        push(@ARGV, '-defines', join(' ', @defines)) if scalar @defines;
        Apache::TestMM::filter_args();
        Apache::TestMM::generate_script('t/TEST');
        Apache::TestMM::generate_script('t/SMOKE');
    };
} else {
    print 'Set TEST_HTTP to enable Apache-Test/HTTP tests', "\n";
    tests 't/*.t';
};

name 'Handel';
license 'perl';
perl_version '5.008001';
all_from 'lib/Handel.pm';

requires 'DBIx::Class'              => '0.08123';
requires 'DBIx::Class::UUIDColumns' => '0.02005';
requires 'DBIx::Class::Validation'  => '0.02005';
requires 'Data::Currency'           => '0.04002';
requires 'Class::Accessor::Grouped' => '0.10006';
requires 'Class::ISA'               => '0.36';
requires 'Class::Inspector'         => '1.24';
requires 'Clone'                    => '0.31';
requires 'Error'                    => '0.17016';
requires 'Carp';
requires 'Locale::Maketext'         => '1.13';
requires 'Module::Pluggable'        => '3.9';
requires 'Module::Starter'          => '1.54';
requires 'Scalar::Util'             => '1.22';
requires 'Tie::Hash'                => '1.03';
requires 'DateTime'                 => '0.61';
requires 'DateTime::Format::MySQL'  => '0.04';
requires 'Locale::Currency'         => '2.07';
requires 'Locale::Currency::Format' => '1.28';
requires 'FormValidator::Simple'    => '0.28';
requires 'Finance::Currency::Convert::WebserviceX' => '0.07001';
requires 'NEXT'                     => '0.64';
requires 'FileHandle'               => '2.01';
requires 'File::Spec::Functions'    => '3.3';
requires 'File::Path'               => '2.08';
requires 'SQL::Translator'          => '0.11006';
requires 'Getopt::Long'             => '2.38';
requires 'Pod::Usage'               => '1.36';
requires 'DBD::SQLite'              => '1.29';
requires 'Data::Currency'           => '0.04002';
build_requires 'Test::More' => '0.94';

feature 'Catalyst Support',
  -default => 0,
  'Catalyst::Runtime'               => '5.80025',
  'Catalyst::Devel'                 => '1.28',
  'Catalyst::View::TT'              => '0.34',
  'Catalyst::Plugin::Session'       => '0.3',
  'Catalyst::Plugin::Session::Store::File'      => '0.18',
  'Catalyst::Plugin::Session::State::Cookie'    => '0.17',
  'YAML'                => '0.71',
  'HTML::FillInForm'    => '2.00';

feature 'AxKit Support',
  -default                   => 0,
  'AxKit' => '1.61';

install_script 'script/handel.pl';

clean_files 'smoke-report-* t/var Handel-* t/TEST t/SMOKE t/logs t/conf/apache_test_config.pm t/conf/extra.conf t/conf/httpd.conf t/conf/modperl_inc.pl t/conf/modperl_startup.pl t/htdocs/index.html t/var';

eval {
    system 'pod2text lib/Handel.pm > README';
};

auto_install;
WriteAll;
